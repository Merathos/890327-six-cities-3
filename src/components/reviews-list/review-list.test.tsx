import * as React from "react";
import * as renderer from "react-test-renderer";
import ReviewList from "./reviews-list";

const reviews = [
  {
    id: `51243`,
    name: `Max`,
    avatar: `img/avatar-max.jpg`,
    rating: 2.5,
    date: `2019-06-13T04:41:20`,
    text: `Consectetur sit velit nulla ipsum. Sunt occaecat amet cupidatat laboris commodo in. Culpa voluptate elit amet officia veniam cillum ad. Ea magna adipisicing laboris incididunt. Laborum eu sit consequat eu nisi. Nulla ipsum ea reprehenderit enim labore do et. Irure anim laborum aute quis.`
  },
  {
    id: `84657`,
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    rating: 3.6,
    date: `2018-05-13T04:41:20`,
    text: `Amet anim Lorem laborum ullamco nisi labore adipisicing ut esse eu. Consectetur laboris consectetur labore ut aute do. Eiusmod sit anim adipisicing excepteur reprehenderit in. Aliqua id laborum nisi veniam dolore cillum esse Lorem non occaecat do mollit.`
  },
  {
    id: `79807`,
    name: `Max`,
    avatar: `img/avatar-max.jpg`,
    rating: 4.7,
    date: `2017-04-13T04:41:20`,
    text: `Aliqua deserunt quis eu culpa nulla. Dolor eiusmod aliquip nulla commodo excepteur ut labore Lorem duis consequat dolor ex aute cillum. Non nisi do eu ex mollit sunt quis cillum fugiat velit et. Aliqua magna excepteur minim adipisicing aute consectetur exercitation laboris laborum sint pariatur in esse consectetur. Elit sit incididunt pariatur est eiusmod non anim deserunt laboris incididunt minim reprehenderit.`
  },
  {
    id: `24357`,
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    rating: 5.8,
    date: `2016-03-13T04:41:20`,
    text: `Velit ipsum voluptate consectetur eiusmod excepteur aliqua et cillum laboris anim nisi dolor. Ex esse id occaecat ex consequat duis sint duis excepteur. Fugiat est nisi dolor officia nisi id. Irure ullamco do ea eiusmod fugiat enim fugiat laborum ea ipsum cupidatat consectetur. Consectetur consequat esse velit tempor commodo ea labore.`
  },
];

it(`Render Review list`, () => {
  const tree = renderer
    .create(<ReviewList
      reviews={reviews}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
