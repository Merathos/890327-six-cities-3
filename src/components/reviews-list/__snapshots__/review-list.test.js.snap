// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Review list 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      4
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "50%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Consectetur sit velit nulla ipsum. Sunt occaecat amet cupidatat laboris commodo in. Culpa voluptate elit amet officia veniam cillum ad. Ea magna adipisicing laboris incididunt. Laborum eu sit consequat eu nisi. Nulla ipsum ea reprehenderit enim labore do et. Irure anim laborum aute quis.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-06-13"
        >
          June 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Angelina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "72%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Amet anim Lorem laborum ullamco nisi labore adipisicing ut esse eu. Consectetur laboris consectetur labore ut aute do. Eiusmod sit anim adipisicing excepteur reprehenderit in. Aliqua id laborum nisi veniam dolore cillum esse Lorem non occaecat do mollit.
        </p>
        <time
          className="reviews__time"
          dateTime="2018-05-13"
        >
          May 2018
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "94%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Aliqua deserunt quis eu culpa nulla. Dolor eiusmod aliquip nulla commodo excepteur ut labore Lorem duis consequat dolor ex aute cillum. Non nisi do eu ex mollit sunt quis cillum fugiat velit et. Aliqua magna excepteur minim adipisicing aute consectetur exercitation laboris laborum sint pariatur in esse consectetur. Elit sit incididunt pariatur est eiusmod non anim deserunt laboris incididunt minim reprehenderit.
        </p>
        <time
          className="reviews__time"
          dateTime="2017-04-13"
        >
          April 2017
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Angelina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "116%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Velit ipsum voluptate consectetur eiusmod excepteur aliqua et cillum laboris anim nisi dolor. Ex esse id occaecat ex consequat duis sint duis excepteur. Fugiat est nisi dolor officia nisi id. Irure ullamco do ea eiusmod fugiat enim fugiat laborum ea ipsum cupidatat consectetur. Consectetur consequat esse velit tempor commodo ea labore.
        </p>
        <time
          className="reviews__time"
          dateTime="2016-03-13"
        >
          March 2016
        </time>
      </div>
    </li>
  </ul>,
]
`;
